app/controllers/ctc/portal/portal_controller.rb

```diff
@@ -16,15 +16,18 @@ def home
     end
   end
 
-  def edit_info; end
+  def edit_info
+    @current_submission = current_intake.client.efile_submissions.last
+  end
 
   def resubmit
     @submission = current_client.efile_submissions.last
-    if @submission.can_transition_to?(:ready_to_resubmit)
-      @submission.transition_to(:ready_to_resubmit)
+    if @submission.can_transition_to?(:resubmitted)
+      current_client.efile_security_informations.create(params.require(:ctc_resubmit_form).permit!)
+      @submission.transition_to(:resubmitted)
```

Comment:
Should this be `!`? In the code as written, we call `.transition_to` which can return false in the case of an error (transition was impossible), but we're ignoring that error, which worries me.

Reply:
it didn't really make sense to me to block the resubmission if there was a problem with the JS, especially since we only added the banner that JS was required on that one page instead of across the app.

Previously, we weren't capturing the efile information here _at all_ (it was copying from the previous submission), so this is at least a step up from that... 

I could put the bang in and then we'll see an error get raised if it does become an issue where we can't get that data and see what happens and fix it if its an issue for people?

Reply:
When I asked about the `!`, I meant that `.transition_to(:resubmitted)` could become `.transition_to!(:resubmitted)`. I still think it's a good idea to either call `transition_to!(...)` or handle the failed transition somehow.

You seem to be talking about the `current_client.efile_security_informations.create(...)` call. I think it's a good idea to make that a `.create!` as well. I do think it's a good idea to block resubmission if JS is not working.

> we only added the banner that JS was required on that one page instead of across the app.

I'm not sure this is true - I think that we add the JS required banner on both pages that require the EfileSecurityInformation during intake; see spec/forms/ctc/confirm_legal_form_spec.rb and spec/forms/ctc/income_form_spec.rb

> I could put the bang in and then we'll see an error get raised if it does become an issue where we can't get that data and see what happens and fix it if its an issue for people?

I'm cool with this!

I'm also cool with no change if you think it's not worth it.